DuckDuckGo/VPNWaitlist.swift

```diff
@@ -0,0 +1,126 @@
+
+        /// Used if the user has been invited via the waitlist and has accepted the Privacy Policy and Terms of Service
+        case waitlistInvited
+
+        /// Used if the user has been invited to test Network Protection directly
+        case inviteCodeInvited
+    }
+
+    static let identifier: String = "vpn"
+    static let apiProductName: String = "networkprotection_ios"
+    static let downloadURL: URL = URL.windows
+
+    static let shared: VPNWaitlist = .init()
+
+    static let backgroundTaskName = "VPN Waitlist Status Task"
+    static let backgroundRefreshTaskIdentifier = "com.duckduckgo.app.vpnWaitlistStatus"
+    static let notificationIdentifier = "com.duckduckgo.ios.vpn.invite-code-available"
+    static let inviteAvailableNotificationTitle = UserText.networkProtectionWaitlistNotificationTitle
+    static let inviteAvailableNotificationBody = UserText.networkProtectionWaitlistNotificationText
+
+    var isAvailable: Bool {
+        isFeatureEnabled
+    }
+
+    var isWaitlistRemoved: Bool {
+        return false
+    }
+
+    let waitlistStorage: WaitlistStorage
+    let waitlistRequest: WaitlistRequest
+    private let privacyConfigurationManager: PrivacyConfigurationManaging
+    private let networkProtectionAccess: NetworkProtectionAccess
+
+    init(store: WaitlistStorage,
+         request: WaitlistRequest,
+         privacyConfigurationManager: PrivacyConfigurationManaging = ContentBlocking.shared.privacyConfigurationManager,
+         networkProtectionAccess: NetworkProtectionAccess = NetworkProtectionAccessController()) {
+        self.waitlistStorage = store
+        self.waitlistRequest = request
+        self.privacyConfigurationManager = privacyConfigurationManager
+        self.networkProtectionAccess = networkProtectionAccess
+
+        let hasWaitlistAccess = privacyConfigurationManager.privacyConfig.isSubfeatureEnabled(NetworkProtectionSubfeature.waitlist)
+        let isWaitlistActive = privacyConfigurationManager.privacyConfig.isSubfeatureEnabled(NetworkProtectionSubfeature.waitlistBetaActive)
+        let internalUser = InternalUserStore()
+        isFeatureEnabled = (hasWaitlistAccess && isWaitlistActive) || internalUser.isInternalUser
+    }
+
+    convenience init(store: WaitlistStorage, request: WaitlistRequest) {
+        self.init(store: store, request: request, privacyConfigurationManager: ContentBlocking.shared.privacyConfigurationManager)
+    }
+
+    var settingsSubtitle: String {
+        switch networkProtectionAccess.networkProtectionAccessType() {
+        case .none:
+            return ""
+        case .waitlistAvailable:
+            return "Join the private waitlist"
+        case .waitlistJoined:
+            return "You're on the list!"
+        case .waitlistInvitedPendingTermsAcceptance:
+            return "You're invited!"
+        case .waitlistInvited, .inviteCodeInvited:
+            assertionFailure("These states should use the VPN connection status")
+            return ""
```

Comment:
These need to move into UserText. The copy here is not final, I was planning to move it once I get the official copy but there's no reason to wait.